### DISCUSSION TITLE
Database connection validation with Quarkus distribution

### DISCUSSION THREAD

**Discussion by devRoemer** [keycloak/keycloak#15750] [Archive]
In HA environments, it was possible before to configure a validation so that the db connection is restored, e.g. after a rolling update of postgres.

It was done like that:

```
<validation>
    <valid-connection-checker class-name="org.jboss.jca.adapters.jdbc.extensions.postgres.PostgreSQLValidConnectionChecker"></valid-connection-checker>
    <background-validation>true</background-validation>
    <background-validation-millis>120000</background-validation-millis>
</validation>
```

More detailed description on the problem can be found here:
  * https://keycloak.discourse.group/t/sql-connection-closed-after-i-o-exception/1876/8
  * https://stackoverflow.com/questions/54865086/keycloak-not-able-to-connect-external-db-after-database-restart

How can the same be accomplished with the Quarkus distribution?

Used Chart for deployment: https://github.com/codecentric/helm-charts/tree/master/charts/keycloakx

**Comment by sschu** [comment]
There are similar options for Quarkus, see https://quarkus.io/guides/datasource#quarkus-agroal_quarkus.datasource.jdbc.background-validation-interval.
